### DISCUSSION TITLE
plot grains bug

### DISCUSSION THREAD

**Discussion by NikitaSakharov** [mtex-toolbox/mtex#842] [Ask Anything] [answered]
Hi everyone, here is the issue I faced with.

I'm trying to plot merged grains and get empty large grains (see attached screen).

Here is code 

%now reconstruct grains
[grains_initial,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[2*degree, 15*degree]);
%delete all of them less then 5 pixels
ebsd(grains_initial(grains_initial.grainSize<5)) = [];
%and now re-reconstruct grains using corrected ebsd data
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[2*degree, 15*degree]);
%now let's work with the boundaries
%extract grain boundaries from all grains
gB = grains.boundary('Copper','Copper');
%define twinning rotation
%define crystal symmetry first
CS = ebsd('Copper').CS;
%define rotation direction
m = Miller(1, 1, 1, CS);
%and define rotation axis
rot = rotation.byAxisAngle(m, 60 * degree);
twinning = rot;
isTwinning = angle(gB.misorientation, twinning) < 5*degree;
%extract twins now
twinBoundary = gB(isTwinning);

%now let's define new grains which consist from the merged twins
[mergedGrains,parentId] = merge(grains,twinBoundary);

plot(mergedGrains,mergedGrains.meanOrientation)
![untitled](https://user-images.githubusercontent.com/62562246/110495857-f9c21d80-8105-11eb-8af5-2f5a7e4ac62b.jpg)

**Comment by kilir** [comment, marked as answer]
Hi Nikita,

actually, the grains are not empty. If you have a closer inspection, it is just, that there is no new mean orientation computed, resp. set to nan.
If you have grain with different parts with a CSL3 relation, what should the mean orientation be? Probably something which is neither the host nor the twin, right?
You can force it by adding 'calcMeanOrientation', however, probably it's better to use e.g. the orientation which occurs most often within a grains (finding the largest areas pre-merging) or something else.

Cheers,
Rüdiger
